pkg/metrics/metrics.go

```diff
@@ -26,30 +26,43 @@ import (
 // Statistic is a common interface for histogram metrics
 type Statistic interface {
 	Collect() prometheus.Collector
-	EvaluateDuration(method string) func()
+	EvaluateDuration(labels prometheus.Labels) func()
+	EvaluateDurationForMethod(method string) func()
+	EvaluateDurationForType(t string) func()
 }
 
+// ExtendedDefBuckets is a default buckets used for csi driver
+var ExtendedDefBuckets = []float64{.025, .05, .1, .5, 1, 3, 5, 10, 15, 30, 45, 90, 180}
```

Comment:
Should this bucket be metrics specific?

Reply:
Yes, buckets settings can be specific for metric. This is just a default.